client/src/partials/purchases/modals/status.js

```diff
@@ -0,0 +1,48 @@
+angular.module('bhima.controllers')
+.controller('PurchaseOrderStatusModalController', PurchaseOrderStatusModalController);
+
+PurchaseOrderStatusModalController.$inject = [
+  '$uibModalInstance', 'NotifyService', 'PurchaseOrderService', 'data'
+];
+
+function PurchaseOrderStatusModalController(Instance, Notify, PurchaseOrder, Data) {
+    var vm = this;
+
+    // global variables 
+    vm.purchase = Data;
+
+    vm.status = vm.purchase.is_confirmed ? 'confirmed' : 'not_confirmed';
+
+    // expose to view 
+    vm.close = Instance.close;
+    vm.submit = submit;
+
+    // submit the choice 
+    function submit(status) {
+        var data;
+
+        switch(status) {
```

Comment:
Can we just do:
```js
var isConfirmed = status === 'confirmed' ? 1 : 0;
var data = { is_confirmed : isConfirmed, is_cancelled: vm.purchase.is_cancled };
updateOrder(vm.purchase.uuid, data);
```
instead of these ~12 lines?